ethash/src/compute.rs

```diff
@@ -233,8 +233,7 @@ fn hash_compute(light: &Light, full_size: usize, header_hash: &H256, nonce: u64)
 
 			Node { bytes: out.assume_init() }
 		},
-		// This is fully initialized before being read, see `let mut compress = ...` below
-		compress_bytes: unsafe { mem::uninitialized() },
+		compress_bytes: [0u8; MIX_WORDS],
```

Comment:
```
❯ cargo bench --features=bench -- compute
bench_light_compute_memmap                                                                            
                        time:   [980.59 us 986.10 us 992.69 us]
                        change: [-3.7744% -1.4892% +0.5632%] (p = 0.20 > 0.05)
                        No change in performance detected.
Found 6 outliers among 100 measurements (6.00%)
  4 (4.00%) high mild
  2 (2.00%) high severe

bench_light_compute_memory                                                                             
                        time:   [1.0336 ms 1.0410 ms 1.0493 ms]
                        change: [-3.9993% -2.0714% +0.0881%] (p = 0.05 > 0.05)
                        No change in performance detected.
Found 9 outliers among 100 measurements (9.00%)
```

Reply:
Amazing, must have been part of one of the LLVM upgrades since this code was written. I absolutely benchmarked it before and 0-initialising it was significantly slower. Wonder if there's anywhere else where we can remove uninit entirely now.